tests/sentry/issues/auto_source_code_config/test_process_event.py

```diff
@@ -612,18 +629,108 @@ def test_very_short_module_name(self) -> None:
                 }
             ],
             platform=self.platform,
+            expected_in_app_stack_trace_rules=["stack.module:a.** +app"],
         )
         assert len(code_mappings) == 1
         assert code_mappings[0].stacktrace_root == "a/"
         assert code_mappings[0].source_path == "src/a/"
 
     def test_handles_dollar_sign_in_module(self) -> None:
-        # No code mapping will be stored, however, we get what would have been created
-        code_mappings, _ = self._process_and_assert_no_configuration_changes(
+        # No code mapping or in-app stack trace rule will be stored
+        code_mappings = self._process_and_assert_no_configuration_changes(
             repo_trees={REPO1: ["src/com/example/foo/Bar.kt"]},
             frames=[{"module": "com.example.foo.Bar$InnerClass", "abs_path": "Bar.kt"}],
             platform=self.platform,
+            # Notice that the in-app stack trace rule does not include "foo" as the stacktrace root does
+            expected_in_app_stack_trace_rules=["stack.module:com.example.** +app"],
```

Comment:
@MichaelSun48 and I discussed this topic and was documented [here](https://www.notion.so/sentry/Derive-code-mappings-in-app-frames-for-Java-Kotlin-Android-1288b10e4b5d8000bc1af92c2837e7ed?pvs=4#19e8b10e4b5d80f6921ddcfffbfd4545).

> For packaged Java, the in-app stack trace rule will have 2 levels of specificity (the reversed domain com.example) while the code mapping will have 3 levels (com/example/foo/). If there are N packages written by the developers (even across various repos), we will immediately mark all of them as in-app.

Concept | Example
-|-
In-app rule | com.example.** (2 levels of specificity)
Code mapping |com/example/foo/ (3 levels of specificity)

This approach will reduce the number of in-app stack trace rules.